internal/producer/src/validation/validation.go

```diff
@@ -79,7 +79,8 @@ func ValidatePending(c *contracts.Contract, pBalance *uint64, pNonce *uint64) er
 	}
 
 	// check for nil sender public key and recip == sha-256 hash of senderPK
-	if c.SenderPubKey == nil || bytes.Equal(c.RecipPubKeyHash, hashing.New(publickey.Encode(c.SenderPubKey))) {
+	recipPKhash := hashing.SHA256Hash{SecureHash: c.RecipPubKeyHash}
```

Comment:
Just FYI I think you can do literals when making structs, so basically these would both be valid:
`recipPKhash := hashing.SHA256Hash{SecureHash: c.RecipPubKeyHash}`
`recipPKhash := hashing.SHA256Hash{c.RecipPubKeyHash}`

If it's multiple fields so long as they're in order it works, but it's just a neat work around I didn't know if you were aware of.